internal/service/models/user.go

```diff
@@ -29,23 +30,23 @@ func NewUser(id string, firstName string, lastName string, email string, age int
 
 func validateUser(user *User) error {
 	if user.id != "" && uuid.Validate(user.id) != nil {
-		return &exceptions.UserValidationErr{Message: "Invalid id"}
+		return &exceptions.UserValidationErr{Err: errors.New("Invalid id")}
 	}
 
 	if user.firstName == "" {
-		return &exceptions.UserValidationErr{Message: "First name is required"}
+		return &exceptions.UserValidationErr{Err: errors.New("First name is required")}
 	}
 
 	if user.lastName == "" {
-		return &exceptions.UserValidationErr{Message: "Last name is required"}
+		return &exceptions.UserValidationErr{Err: errors.New("Last name is required")}
 	}
 
 	if user.email == "" {
-		return &exceptions.UserValidationErr{Message: "Email is required"}
+		return &exceptions.UserValidationErr{Err: errors.New("Email is required")}
 	}
 
 	if user.age < 18 {
-		return &exceptions.UserValidationErr{Message: "User must be at least 18 years old"}
+		return &exceptions.UserValidationErr{Err: errors.New("User must be at least 18 years old")}
 	}
 
 	return nil
```

Comment:
This should return a list of validation errors so that if there are multiple problems with the user they are sent back at once.

Reply:
This is a second validation for the user, I have the bean validation in the handler, but if for some reason, another endpoint or any other kind of entry point in the application, I have this validation for the domain model. The bean validation returns all errors at once, this have other purpose validation, it's more field by field